components/collapsible/collapsible.test.js

```diff
@@ -115,18 +145,24 @@ describe('Dialtone vue Collapsible Component Tests', function () {
         await wrapper.setProps({ open: true });
       });
 
-      it('content is expanded', function () {
-        assert.isTrue(contentElement.isVisible());
+      it('content is expanded when open = true', function () {
+        assert.isTrue(contentOnExpandedElement.isVisible());
       });
 
-      it('clicking does not collapse content', async function () {
+      it('clicking does NOT collapse content', async function () {
         await anchorElement.trigger('click');
-        assert.isTrue(contentElement.isVisible());
+        contentOnExpandedElement = wrapper.find('[data-qa="content-on-expanded-element"]');
+        contentOnCollapsedElement = wrapper.find('[data-qa="content-on-collapsed-element"]');
+        assert.isTrue(contentOnExpandedElement.isVisible());
+        assert.isFalse(contentOnCollapsedElement.exists());
       });
 
       it('updating open prop does collapse content', async function () {
         await wrapper.setProps({ open: false });
```

Comment:
Setting state should be done in describes, ideally 'it' is just an assertion if possible. More on this here: https://vue.dialpad.design/?path=/story/docs-component-driven-development-unit-tests--page

This goes for most of the tests in this PR. It is far more maintainable and scalable to write tests in this way.

Example:

```
describe('When open prop is set to true', function () {
  describe('When anchor is clicked', function () {
    // beforeEach here to perform click
    it('does not collapse content', function () {
      // assertion
    });
  });
});
```

Reply:
Sure, but one issue I noticed with this approach is when multiple user actions are needed for a specific test. For example, for one test, I wanted to test the user clicking the anchor multiple times and perform assertions based on that. I usually did that sequentially in an `it` block, like so:

```js
it('test', async function () {
         await anchorElement.trigger('click');
        // test stuff
        await anchorElement.trigger('click'); // click again
       // test more stuff
})
```

In a describe, it would look a bit awkward:

```js
describe('when user clicks anchor when it is collapsed', function () {
      beforeEach(async function () {
        // need to trigger twice since the initial state of the
        // wrapper is expanded. We want it to be collapsed initially.
        await anchorElement.trigger('click');
        await anchorElement.trigger('click');
      });
      it('should be expanded', function () {
        findAllElements();
        assert.isTrue(contentOnExpandedElement.isVisible());
        assert.isFalse(contentOnCollapsedElement.exists());
      });
    });

```

Reply:
You do have a way of setting the initial expanded state via the `open` prop. I'd recommend just using that to render the desired initial state and then clicking to test the trigger. Alternatively you could have a nested describe for each click and have one test to check whether it's collapsed, and one test to check whether it's expanded.

Reply:
I opted for the latter, the thing with this test is that we're trying to make sure that the behaviour works correctly when the `open` prop is undefined, so it wouldn't work if we explicitly set the `open` prop here.

I just nested the `describe`s as you mentioned and it works.